lib/signaling/v2/peerconnection.js

```diff
@@ -101,12 +101,16 @@ class PeerConnectionV2 extends StateMachine {
     const peerConnection = new RTCPeerConnection(configuration, options.chromeSpecificConstraints);
     const sdpFormat = getSdpFormat(options.sdpSemantics);
 
-    const localMediaStream = isFirefox && RTCPeerConnection.prototype.addTransceiver
+    const localMediaStream = sdpFormat === 'unified' && RTCPeerConnection.prototype.addTransceiver
       ? null
       : new options.MediaStream();
 
     if (options.dummyAudioMediaStreamTrack) {
-      peerConnection.addTrack(options.dummyAudioMediaStreamTrack, localMediaStream || new MediaStream());
+      if (localMediaStream) {
+        peerConnection.addTrack(options.dummyAudioMediaStreamTrack, localMediaStream);
+      } else {
+        peerConnection.addTransceiver(options.dummyAudioMediaStreamTrack);
```

Comment:
I am not sure this is a safe refactor. If `offerToReceiveAudio` is true, `addTrack` and `addTransceiver` will yield different results. `addTrack` will reuse the media section introduced by `offerToReceiveAudio`, whereas `addTransceiver` will not. Since this code block deals with the `dummyAudioMediaStreamTrack`, which we intentionally want to be assigned to the first media section, I think we should stick with `addTrack`.